sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -114,6 +110,35 @@ object GpuFileFormatDataWriter {
       }
     }
   }
+
+  // All data is sorted ascending with default null ordering
+  private val nullsSmallest = Ascending.defaultNullOrdering == NullsFirst
+
+  // distinct value sorted the same way the input data is sorted.
+  private[rapids] def distinctAndSort(t: Table): Table = {
+    val columnIds = 0 until t.getNumberOfColumns
+    withResource(t.groupBy(columnIds: _*).aggregate()) { distinct =>
+      distinct.orderBy(columnIds.map(OrderByArg.asc(_, nullsSmallest)): _*)
+    }
+  }
+
+  // Get the split indexes for t given the keys we want to split on
+  private[rapids] def splitIndexes(t: Table, keys: Table): Array[Int] = {
```

Comment:
nit: The exact same name is used in other places in this file to store the data. Could we try and find a better way to differentiate the two?

Reply:
I moved it back into the GpuDynamicPartitionDataSingleWriter, not an utils function any more.